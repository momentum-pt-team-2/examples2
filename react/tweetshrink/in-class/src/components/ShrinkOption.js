import React from 'react'

class ShrinkOption extends React.Component {
  render () {
    const { onOptionChange, id, label } = this.props
    return (
      <div className='col-6'>
        <label htmlFor={id}>
          <input type='checkbox' id={id}
            onChange={(event) => onOptionChange(id, event)} />
          <span>{label}</span>
        </label>
      </div>
    )
  }
}

export default ShrinkOption
