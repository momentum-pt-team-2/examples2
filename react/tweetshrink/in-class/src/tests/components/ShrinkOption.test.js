/* globals test, expect */

import React from 'react'
import ReactDOM from 'react-dom'
import ShrinkOption from '../../components/ShrinkOption'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

test('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<ShrinkOption />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('does not change inadvertenly', () => {
  const component = renderer.create(
    <ShrinkOption id={'spaces'} label={'Eliminate extra spaces'} onOptionChange={() => null} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders a checkbox and shows label', () => {
  const wrapper = shallow(<ShrinkOption id={'spaces'} label={'Eliminate extra spaces'} />)
  expect(wrapper.find('input[type="checkbox"]')).toHaveLength(1)
  expect(wrapper.text()).toContain('Eliminate extra spaces')
})
