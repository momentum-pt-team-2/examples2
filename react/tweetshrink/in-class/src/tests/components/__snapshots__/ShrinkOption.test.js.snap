// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not change inadvertenly 1`] = `
<div
  className="col-6"
>
  <label
    htmlFor="spaces"
  >
    <input
      id="spaces"
      onChange={[Function]}
      type="checkbox"
    />
    <span>
      Eliminate extra spaces
    </span>
  </label>
</div>
`;
